### ISSUE TITLE
Use Etags / Last-Modified Header for GH API calls

### ISSUE THREAD

**Issue by amercader** [frictionlessdata/goodtables.io#156]
As this do not count towards the API allowance. Specially when syncing as we do lots of requests.

pygithub3 seems to have some sort of support for it but we can also use direct calls if necessary.

**Comment by roll** [comment]
@amercader 
Yes it supported and I've discovered it on first place. It works good but one problem that this mechanism requires to persist all etags on our side. So for example an user have 500 repos so there is a need of 500 hooks iterator requests. All of this request will be having some etag we need to sore somewhere.

**Comment by amercader** [comment]
I was thinking a simple key/value store on redis with the hashed url and the etag
